hail/python/hailtop/hailctl/batch/initialize.py

```diff
@@ -0,0 +1,252 @@
+                         verbose: bool) -> Tuple[Optional[str], str, bool]:
+    from .utils import check_for_gcloud  # pylint: disable=import-outside-toplevel
+    assert len(supported_regions) > 0
+
+    gcloud_installed = await check_for_gcloud()
+    if not gcloud_installed:
+        typer.secho('Have you installed gcloud? For directions see https://cloud.google.com/sdk/docs/install '
+                    'To log into gcloud run:\n'
+                    '> gcloud auth application-default login',
+                    fg=typer.colors.RED)
+        raise Abort()
+
+    create_remote_tmpdir = Confirm.ask('Do you want to create a new bucket in project for temporary files generated by Hail?')
+    if create_remote_tmpdir:
+        remote_tmpdir, location, warnings = await setup_new_remote_tmpdir(
+            supported_regions=supported_regions,
+            username=username,
+            service_account=hail_identity,
+            verbose=verbose,
+        )
+    else:
+        remote_tmpdir, location, warnings = await setup_existing_remote_tmpdir(hail_identity, verbose)
+
+    return (remote_tmpdir, location, warnings)
+
+
+async def async_basic_initialize(domain: Optional[str] = None, verbose: bool = False):
+    from hailtop.auth import async_get_userinfo  # pylint: disable=import-outside-toplevel
+    from hailtop.batch_client.aioclient import BatchClient  # pylint: disable=import-outside-toplevel
+    from hailtop.config.deploy_config import get_deploy_config  # pylint: disable=import-outside-toplevel
+    from hailtop.hailctl.config.cli import set as set_config, list as list_config  # pylint: disable=import-outside-toplevel
+
+    from .utils import already_logged_into_service, login_to_service  # pylint: disable=import-outside-toplevel
+
+    already_logged_in = await already_logged_into_service(domain)
+    if not already_logged_in:
+        typer.secho('You are not currently logged in to Hail. Redirecting you to a login screen.', fg=typer.colors.YELLOW)
+        await login_to_service(domain)
+        typer.secho('In the future, you can use `hailctl auth login` to login to Hail.', fg=typer.colors.YELLOW)
+    else:
+        domain = get_deploy_config()._domain
+        typer.secho(f'You are currently logged in to Hail at domain {domain}.')
+
+    user_info = await async_get_userinfo()
+    username = user_info['username']
+    hail_identity = user_info['hail_identity']
+    trial_bp_name = user_info['trial_bp_name']
+
+    batch_client = await BatchClient.create(trial_bp_name)
+
+    async with batch_client:
+        cloud = await batch_client.cloud()
+        supported_regions = await batch_client.supported_regions()
+
+        if cloud == 'gcp':
+            remote_tmpdir, tmpdir_region, warnings = await initialize_gcp(username, hail_identity, supported_regions, verbose)
+        else:
+            remote_tmpdir = Prompt.ask('Enter a path to an existing remote temporary directory (ex: gs://my-bucket/batch/tmp)')
```

Comment:
This else block seems to assume azure, so let's use an azure dummy url instead of a GCS one